probe_src/python/probe_py/manual/cli.py

```diff
@@ -146,5 +195,539 @@ def dump(
+        result = subprocess.run(ssh_command, capture_output=True, text=True, check=True)
+        xdg_data_home = result.stdout.strip()
+        return xdg_data_home
+    except subprocess.CalledProcessError as e:
+        print(f"Error retrieving XDG_DATA_HOME: {e}")
+        return None
+
+def get_remote_home_env(remote_user: str, remote_host: str, port: int)->str | None:
+    try:
+        ssh_command = [
+            "ssh",
+            "-p",
+            str(port),  # Specify the port
+            f"{remote_user}@{remote_host}",  # Remote user and host
+            "echo $HOME",  # Command to retrieve HOME
+        ]
+        result = subprocess.run(ssh_command, capture_output=True, text=True, check=True)
+        home = result.stdout.strip()
+        return home
+    except subprocess.CalledProcessError as e:
+        print(f"Error retrieving XDG_DATA_HOME: {e}")
+        return None
+
+def get_file_info_on_local(file_path:pathlib.Path)->tuple[InodeVersion, InodeMetadataVersion]:
+    stat_info = os.stat(file_path)
+    device_major = os.major(stat_info.st_dev)
+    device_minor = os.minor(stat_info.st_dev)
+    inode_val = stat_info.st_ino
+    hostname = socket.gethostname()
+    ip_address = socket.gethostbyname(hostname)
+    packed_ip = socket.inet_aton(ip_address)
+    host = struct.unpack("!I", packed_ip)[0]
+
+    # c uses mtime in nanoseconds
+    mtime_micro_seconds = stat_info.st_mtime
+    mtime = int(mtime_micro_seconds * 1_000_000_000)
+
+    size = stat_info.st_size
+    inode = Inode(host, device_major, device_minor, inode_val)
+    inode_version = InodeVersion(inode, mtime, size)
+    stat_results = os.stat(file_path)
+    serialized_stat_results = pickle.dumps(stat_results)
+    inode_metadata = InodeMetadataVersion(
+        inode_version, serialized_stat_results
+    )
+    return inode_version, inode_metadata
+
+
+def get_file_info_on_remote(remote_host:str, file_path:pathlib.Path, remote_user:str, port:int) -> tuple[InodeVersion, InodeMetadataVersion]:
+    command = [
+        "ssh",
+        f"-p {port}",
+        f"{remote_user}@{remote_host}",
+        f'stat -c "%D %i %s %Y" {file_path}',
+    ]
+
+    process = subprocess.run(command, check=True, capture_output=True, text=True)
+
+    output = process.stdout.strip()
+    device_hex, str_inode, str_size, str_mtime = output.split()
+
+    inode_val = int(str_inode)
+    size = int(str_size)
+    mtime = int(str_mtime) * 1_000_000_000
+
+    # Get host information
+    hostname = socket.gethostname()
+    ip_address = socket.gethostbyname(hostname)
+    packed_ip = socket.inet_aton(ip_address)
+    host = struct.unpack("!I", packed_ip)[0]
+
+    command = [
+        "ssh",
+        f"-p {port}",
+        f"{remote_user}@{remote_host}",
+        f'python3 -c \'import os, json; stat_info = os.stat("{file_path}"); device_id = stat_info.st_dev; result = {{"device_major": os.major(device_id), "device_minor": os.minor(device_id)}}; print(json.dumps(result))\'',
```

Comment:
Why do we need this? Stat has major and minor device as fields:

```
$ stat --help
...
  %Hd  major device number in decimal
  %Ld  minor device number in decimal
...
```

Reply:
@charmoniumQ I tried using `f'stat -c "device_major: %t\ndevice_minor: %T\n" {file_path}'` instead but it gave me value 0 for both `device-major` and `device-minor`

Reply:
@charmoniumQ